### PR TITLE
Implementation of Suggestions Section into the Rentals Screen

### PR DESCRIPTION
**Pull request by joaquinbozzalla**
## Summary ##

**Trello card**: https://trello.com/c/PjJSyNnV

Implementation of a horizontal scrollable collection at the bottom of the rentals screen with suggested books.

### iPhone 11

simulator_screenshot_17C8EFC0-9B7D-4327-A421-6D860E36262B

### iPhone SE

simulator_screenshot_67F843B8-A233-4125-A983-A3F5466D8341

## Screencast ##

demo4

### CODE REVIEW

File: WBooks/BookDetails/CommentsSection/CommentsSectionViewController.swift [hunk 1]
```diff
@@ -46,10 +46,9 @@ final class CommentsSectionViewController: UIViewController {
     }
     
     private func reloadTable() {
+        commentsSectionView.tableComments.reloadData()
```
Comment (carcos-wolox) [inline WBooks/BookDetails/CommentsSection/CommentsSectionViewController.swift]: shouldn't this be in the else of `viewModel.noComments` ?

File: WBooks/Rentals/RentalsViewController.swift [hunk 2]
```diff
@@ -10,6 +10,7 @@ import UIKit
 final class RentalsViewController: UIViewController {
     private let rentalsView = RentalsView()
     private var tableViewController = BooksTableViewController(viewModel: RentalsViewModel())
```
Comment (carcos-wolox) [inline WBooks/Rentals/RentalsViewController.swift]: Are you changing this `tableViewController`? or can be `let`?

File: WBooks/Rentals/RentalsViewController.swift [hunk 3]
```diff
@@ -19,6 +20,24 @@ final class RentalsViewController: UIViewController {
...
+        rentalsView.suggestionsContainer.addSubview(suggestionsController.view)
+        
+        rentalsView.suggestionsContainer.translatesAutoresizingMaskIntoConstraints = false
+        suggestionsController.view.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            rentalsView.suggestionsContainer.topAnchor.constraint(equalTo: suggestionsController.view.topAnchor),
+            rentalsView.suggestionsContainer.bottomAnchor.constraint(equalTo: suggestionsController.view.bottomAnchor),
+            suggestionsController.view.leadingAnchor.constraint(equalTo: rentalsView.suggestionsContainer.leadingAnchor, constant: 15.0),
+            suggestionsController.view.trailingAnchor.constraint(equalTo: rentalsView.suggestionsContainer.trailingAnchor, constant: -15.0),
+        ])
```
Comment (carcos-wolox) [inline WBooks/Rentals/RentalsViewController.swift:41]: couldn't you use `fixIn` here?

File: WBooks/Rentals/Suggestions/SuggestionsViewController.swift [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    
+    private func loadData() {
+        viewModel.fecthSuggestions(
+            onSuccess: { [weak self] in self?.reloadCollection() },
+            onError: { [weak self] _ in self?.showMessageIntoSuggestionsSection(message: "SUGGESTIONS_ERROR_MESSAGE".localized()) }
+        )
+    }
+    
+    private func reloadCollection() {
+        suggestionsView.suggestionsCollection.reloadData()
```
Comment (carcos-wolox) [inline WBooks/Rentals/Suggestions/SuggestionsViewController.swift]: shouldn't this be in an `else` ?

**Review by carcos-wolox — APPROVED** [review]
So gooodd!! your last functionality-PR 🚀 💪

File: WBooks/Rentals/RentalsViewController.swift (hunk 3 above)
Comment (joaquinbozzalla) [reply WBooks/Rentals/RentalsViewController.swift:41]: The code is similar but the constraints are slightly different

**Pull request merged**